src/adapters/rhythmone.js

```diff
@@ -0,0 +1,360 @@
+w.trackR1Impression = track;
+  
+module.exports = function(bidManager, global, loader){
+
+  var version = "0.9.0.0",
+    defaultZone = "1r",
+    defaultPath = "mvo",
+    bidfloor = 0,
+    currency = "USD",
+    debug = false,
+    auctionEnded = false,
+    requestCompleted = false,
+    placementCodes = {};
+    
+  if(typeof global === "undefined")
+    global = window;
+      
+  if(typeof bidManager === "undefined")
+    bidManager = bidmanager;
+    
+  if(typeof loader === "undefined")
+    loader = ajax;
+  
+  //track(debug, 'hb', 'start', version);
+  
+  function applyMacros(txt, values){
+    return txt.replace(/\{([^\}]+)\}/g, function(match){
+      var v = values[match.replace(/[\{\}]/g, "").toLowerCase()];
+      if(typeof v !== "undefined") return v;
+      return match;
+    });
+  }
+  
+  function load(bidParams, url, postData, callback){
+    if(bidParams.method === "get"){
+      loader(url, function(responseText, response){
+        if(response.status === 200)
+          callback(200, "success", response.responseText);
+        else
+          callback(-1, "http error "+response.status, response.responseText);
+      }, false, {method:"GET"});
+    }
+    else{
+      loader(url, function(responseText, response){
+        if(response.status === 200)
+          callback(200, "success", response.responseText);
+        else
+          callback(-1, "http error "+response.status, response.responseText);
+      }, postData, {method:"POST", contentType: "application/json"});
+    }
+  }
+
+  var bidderCode = "rhythmone",
+    bidLostTimeout = null;
+  
+  function setIfPresent(o, key, value){
+    try{
+      if(typeof value === "function")
+        o[key] = value();
+    }catch(ex){}
+  }
+  
+  function logToConsole(txt){
+    if(debug)
+      console.log(txt);
+  }
+  
+  function sniffAuctionEnd(){
+  
+    global.$$PREBID_GLOBAL$$.onEvent('bidWon', function (e) {
+    
+      if(e.bidderCode === bidderCode){
+        placementCodes[e.adUnitCode] = true;
+        track(debug, 'hb', "bidWon");
+      }
+      
+      if(auctionEnded){
+        clearTimeout(bidLostTimeout);
+        bidLostTimeout = setTimeout(function(){
+          for(var k in placementCodes)
+            if(placementCodes[k] === false)
+              track(debug, 'hb', "bidLost");
+        }, 50);
+      }
+    });
+  
+    global.$$PREBID_GLOBAL$$.onEvent('auctionEnd', function () {
+    
+      auctionEnded = true;
+
+      if(requestCompleted === false)
+        track(debug, 'hb', 'rmpReplyFail', "prebid timeout post auction");
+    });
+  }
+  
+  function getBidParameters(bids){
+    for(var i=0;i<bids.length;i++)
+      if(typeof bids[i].params === "object" && bids[i].params.placementId)
+        return bids[i].params;
+    return null;
+  }
+  
+  function noBids(params){
+    for(var i=0; i<params.bids.length; i++){
+      if(params.bids[i].success !== 1){
+        var bid = bidfactory.createBid(2);
```

Comment:
Adapters can now pass a constant for status and the originating bid request, the first for readability and the second to match bid request / response IDs per #509.

e.g. 
```
var bid = bidfactory.createBid(CONSTANTS.STATUS.NO_BID, bidRequest);
```